### PR TITLE
feat: integrate signup form on the main website

### LINKED ISSUE #224: Complete ilmomasiina integration
100% ilmomasiina integration, never leave the website for another for ilmomasiina. Does not include admin panel for ilmomasiina or email edits

- [ ] error handling
- [ ] all corner cases
- [ ] stress test critical path (ilmo button)

### PR DESCRIPTION
**Pull request by MikaelSiidorow**
## Description

- implement full sign up integration on the main website:
  - submit initial sign up
  - edit existing sign ups
  - delete sign up 
- add ui library components for styled form inputs
  - base input
  - checkbox
  - radio
  
Closes #224 

### Before submitting the PR, please make sure you do the following

- [x] If your PR is related to a previously discussed issue, please [link to it](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) here.
- [x] Prefix your PR title with feat:, fix:, chore:, or docs:.
- [x] This message body should clearly illustrate what problems it solves.
- [x] Make sure the commit history is linear, up-to-date with main branch and does not contain any erroneous changes

### Formatting and linting

- [x] Format code with `pnpm format` and lint the project with `pnpm lint`

### CODE REVIEW

File: apps/web/src/locales/en.ts [hunk 1]
```diff
@@ -31,11 +31,34 @@ const en = {
...
+    "You are in the quota {quotaName} at position {position}/{quotaSize}.",
+  "ilmomasiina.form.optional": "optional",
+  "ilmomasiina.form.Shown in the public list of sign ups":
+    "Shown in the public list of sign ups",
+  "ilmomasiina.form.Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}":
+    "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}",
+  "ilmomasiina.form.First name": "First name",
+  "ilmomasiina.form.Last name": "Last name",
+  "ilmomasiina.form.Email": "Email",
+  "ilmomasiina.form.Show name in the public list of sign ups": "Email",
```
Comment (kahlstrm) [inline apps/web/src/locales/en.ts:55]:
```suggestion
  "ilmomasiina.form.Show name in the public list of sign ups": "Show name in the public list of sign ups",
```

File: apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+          signupEditToken={signupEditToken}
+          signup={signupInfo.data.signup}
+          event={signupInfo.data.event}
+          saveAction={saveSignUpAction}
+          deleteAction={deleteSignUpAction}
+        />
+        <footer className="text-sm text-gray-800">
+          <p>
+            {t(
+              "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}",
```
Comment (kahlstrm) [inline apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx:88]: This currently makes the link possibly wrap across lines and not act as an actual link, i.e. `anchor.`
How about the following diff (can be applied with `git apply -`):
```shell
diff --git a/apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx b/apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx
index d07b79e..cbc2560 100644
--- a/apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx
+++ b/apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx
@@ -85,11 +85,16 @@ export default async function Page({
         <footer className="text-sm text-gray-800">
           <p>
             {t(
-              "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}",
-              {
-                eventUrl: `${process.env.PUBLIC_ILMOMASIINA_URL ?? ""}/signup/${signupId}/${signupEditToken}`,
-              },
-            )}
+              "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page",
+            )}{" "}
+            <a
+              className="font-semibold underline"
+              href={`${process.env.PUBLIC_ILMOMASIINA_URL ?? ""}/signup/${signupId}/${signupEditToken}`}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              {t("here")}
+            </a>
           </p>
         </footer>
       </div>
diff --git a/apps/web/src/locales/en.ts b/apps/web/src/locales/en.ts
index 250aa03..a4a2598 100644
--- a/apps/web/src/locales/en.ts
+++ b/apps/web/src/locales/en.ts
@@ -47,8 +47,9 @@ const en = {
   "ilmomasiina.form.optional": "optional",
   "ilmomasiina.form.Shown in the public list of sign ups":
     "Shown in the public list of sign ups",
-  "ilmomasiina.form.Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}":
-    "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}",
+  "ilmomasiina.form.Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page":
+    "Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page",
+  "ilmomasiina.form.here": "here",
   "ilmomasiina.form.First name": "First name",
   "ilmomasiina.form.Last name": "Last name",
   "ilmomasiina.form.Email": "Email",
diff --git a/apps/web/src/locales/fi.ts b/apps/web/src/locales/fi.ts
index 6be7f39..365f54f 100644
--- a/apps/web/src/locales/fi.ts
+++ b/apps/web/src/locales/fi.ts
@@ -47,8 +47,9 @@ const fi = {
   "ilmomasiina.form.optional": "valinnainen",
   "ilmomasiina.form.Shown in the public list of sign ups":
     "Näytetään julkisessa osallistujalistassa",
-  "ilmomasiina.form.Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page: {eventUrl}":
-    "Huomio: Ilmoittautuminen suoraan sivuilla on beta-vaiheessa, jos kohtaat ongelmia voit ilmoittautua tapahtuman sivulla ilmomasiinassa: {eventUrl}",
+  "ilmomasiina.form.Note: Sign up integration is in beta, if you encounter any issues you can sign up directly on the event page":
+    "Huomio: Ilmoittautuminen suoraan sivuilla on beta-vaiheessa, jos kohtaat ongelmia voit ilmoittautua tapahtuman sivulla ilmomasiinassa",
+  "ilmomasiina.form.here": "täällä",
   "ilmomasiina.form.First name": "Etunimi",
   "ilmomasiina.form.Last name": "Sukunimi",
   "ilmomasiina.form.Email": "Sähköposti",
```
Not the prettiest and finnish text is a bit clunky, but you get the gist 😄

Comment (kahlstrm) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx:88]: Also, I wonder if we could also link to the "Give feedback" here somehwere as well?

File: apps/web/src/lib/api/external/ilmomasiina/actions.ts [hunk 3]
```diff
@@ -32,5 +43,143 @@ export async function signUp(formData: FormData) {
...
+      answer,
+    })),
+    language: locale,
+    firstName,
+    lastName,
+    email,
+    namePublic: namePublic === "on",
+  });
+
+  if (!response.ok) {
```
Comment (kahlstrm) [inline apps/web/src/lib/api/external/ilmomasiina/actions.ts:120]: Noticed while testing that in the ilmomasiina side while editing a filled out entry, changing name/email is not allowed (expected behavior I think), but when an edit with these fields is sent from the client and next forwards it to ilmomasiina, ilmomasiina responds with a 200 response instead of detecting the invalid changes for the name/email fields. It doesn't update these fields, but doesn't send a 400 with a validation error 😅 

Bug with Ilmomasiina, but we should regardless disable editing the name & email fields on a confirmed signup.

File: apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        <p className="w-full max-w-sm">
+          {t(
+            "You can edit your sign up or delete it later from this page, which will be sent to your email in the confirmation message",
+          )}
+        </p>
+        <SubmitButton />
+        <Button
+          formNoValidate
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises -- server actions can be ignored promises
+          formAction={deleteAction}
```
Comment (kahlstrm) [inline apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx:276]:
```suggestion
          formAction={void deleteAction}
```

Comment (kahlstrm) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx:276]: Also I wonder if we could add the same "are you sure" functionality here as there is with the original ilmomasiina form? To prevent accidental cancellations.

**Review by kahlstrm — COMMENTED** [review]
Looks mostly good, apart from the incorrect localization and some improvements ideas.

File: apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx (hunk 2 above)
Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx:88]: There might be better support for links (or HTML/JSX) in the i18n lib, would need to check. I don't think this is too great since the link text "here" is not good.

This was more meant as a temporary thing for maybe 1-2 sign up at most, that's why I didn't bother too much with making it a proper link since it works 😅

File: apps/web/src/lib/api/external/ilmomasiina/actions.ts (hunk 3 above)
Comment (MikaelSiidorow) [reply apps/web/src/lib/api/external/ilmomasiina/actions.ts:120]: Good catch, I recall thinking about this but then forgot

File: apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx (hunk 4 above)
Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx:276]: The double button click behavior of ilmomasiina is a bit odd, we had a design for a confirmation modal/popover by @ellimattila that we could implement here

Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx:276]: I don't think `void deleteActioṉ` works here, but I could be wrong. I recall last time I tried it, it broke the server action completely

File: apps/web/src/locales/en.ts (hunk 1 above)
Comment (MikaelSiidorow) [reply apps/web/src/locales/en.ts]: Fixed

File: apps/web/src/lib/api/external/ilmomasiina/actions.ts (hunk 3 above)
Comment (MikaelSiidorow) [reply apps/web/src/lib/api/external/ilmomasiina/actions.ts:120]: Fixed

File: apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx (hunk 2 above)
Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/page.tsx:88]: Won't fix, will just remove the notice after is validated working on Monday (Nettisivusitsit signup)
